src/main/java/org/spdx/sbom/gradle/SpdxSbomPlugin.java

```diff
@@ -117,9 +117,17 @@ private void createTaskForTarget(
                 SpdxSbomTask.class,
                 t -> {
                   t.setGroup("Spdx sbom tasks");
-                  t.getOutputDirectory().set(project.getLayout().getBuildDirectory().dir("spdx"));
+                  if (target.getOutputFile().isPresent()) {
```

Comment:
This causes the value of `target.outputFile` to be read as soon as the task is instantiated, right?

It could be nice to make it lazy so that `target.outputFile` is queried only when it's needed, in case someone accidentally instantiates the the task and then later changes the value of `Target.outputFile`

I think this can be done by moving the check inside `Provider.map` ( https://docs.gradle.org/current/javadoc/org/gradle/api/provider/Provider.html#map-org.gradle.api.Transformer- )

to get something sort of like `t.outputFile.set(target.outputFile.map { f -> if f.isPresent() f else "$buildDir/spdx/${target.name}.spdx.json"}`

Reply:
oh yeah maybe, I thought it was sufficiently lazy... lemme play around with it a bit.

Reply:
oh I should've just applied this on the extension convention earlier in the plugin init. I'll upload that change